src/symbol.c

```diff
@@ -691,4 +698,5 @@ mrb_init_symbol(mrb_state *mrb)
   mrb_define_method_id(mrb, sym, MRB_SYM(to_sym),  mrb_obj_itself,  MRB_ARGS_NONE());      /* 15.2.11.3.4 */
   mrb_define_method_id(mrb, sym, MRB_SYM(inspect), sym_inspect, MRB_ARGS_NONE());          /* 15.2.11.3.5(x) */
   mrb_define_method_id(mrb, sym, MRB_OPSYM(cmp),   sym_cmp,     MRB_ARGS_REQ(1));
+  mrb_define_method_id(mrb, sym, MRB_OPSYM(eq),    sym_eq,      MRB_ARGS_REQ(1));
```

Comment:
I don't think this `sym_eq` change would improve the performance, since `mrb_obj_equal_m` (that defines `Object#==`) has exactly the same implementation. Should be removed from the PR.

Reply:
`Symbol` inherits the `==` method from `Comparable` which does a lexical comparison. a new commit will override `==` to use `mrb_obj_equal_m`. (or should i make a separate pr for this?)